pkg/virt-handler/vm.go

```diff
@@ -104,6 +104,14 @@ const (
 	unableCreateVirtLauncherConnectionFmt = "unable to create virt-launcher client connection: %v"
 )
 
+var (
+	supportedMACAnnotations = []string{
+		"cni.cilium.io/macAddrs",           // cilium
+		"ovn.kubernetes.io/mac_address",    // kube-ovn
+		"cni.projectcalico.org/hwAdddress", // calico
```

Comment:
I'm concerned about this - afaiu, all the KubeVirt is CNI agnostic - it simply does not care who / what configures the pod interface - it just is. 

This would break that.

Reply:
For now it is just MVP.

I have in plans to write KEP to add common functionality for requesting MAC-address from CNI into Kubernetes project. But this will lead a need to patch Kubernetes and containerd code + all CNI plugins which does not support this yet.

Possible this will be easier in case of implementig kubernetes/kubernetes#69882

Here are some notes:
https://kubernetes.slack.com/archives/C09QYUH5W/p1651655504890309

Reply:
Another option is add extra logic to compare old and new MAC-addresses, and
- **if they are different**: to detach and attach NIC 
- **if they are same**: just link down/link up

This way we will not depend on CNI specific annotations.

Reply:
I would favor any CNI agnostic solution.